### PR TITLE
syncer(dm): merge feature branch

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5834

### What is changed and how it works?

this PR includes new handle-error logic

#### Tests 

 - pass original test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by buchuitoudegou — APPROVED** [review]

**Review by D3Hunter — APPROVED** [review]

File: dm/syncer/syncer.go [hunk 1]
```diff
@@ -1939,14 +1940,19 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
...
+		if gset == nil {
+			gtidSet, err2 := gtid.ParserGTID(s.cfg.Flavor, currentGTID)
+			if err2 != nil {
+				return err2
+			}
+			return currentLocation.SetGTID(gtidSet)
 		}
-		err2 = currentLocation.SetGTID(newGTID)
+
+		currEndGSet = currentLocation.GetGTID().Clone()
```
Comment (GMHDBJD) [inline dm/syncer/syncer.go:1952]: Will this update currentLocation.GTID?

**Pull request merged**